### PR TITLE
Set ELF SONAME using `rust-android-gradle` plugin.

### PR DESCRIPTION
**Pull request by ncalexan**
This follows-up on #248 and should address #197.

### CODE REVIEW

**Comment by ncalexan** [comment]
@tcsc you're the best person to review this.  https://github.com/mozilla/application-services/pull/250/commits/28d00d1ce191e4adab54cef0034ca30e149b1317 is the relevant commit, but the real work is over in https://github.com/ncalexan/rust-android-gradle/commit/c52f5e26b2869dd5c5760dfd8a181cddcb80663c#diff-32284e6eba45e43ade5f5273691ebce5R83.

It seems to produce a good `.so` (and AAR); I'm testing on some Android emulators now.

File: fxa-client/sdks/android/library/build.gradle [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    targetDirectory = '../../../../target'
+
+    libname = 'libfxa_client'
+
+    // The Cargo targets to invoke.  The mapping from short name to target
+    // triple is defined by the `rust-android-gradle` plugin.
+    targets = [
+        // 'default', // TODO: For unit tests.
+        'arm',
+        // 'arm64', // XXX: Just for development speed.
```
Comment (vladikoff) [inline fxa-client/sdks/android/library/build.gradle]: need to uncomment this before merge?

Can gradle have a task that for each env, so in development you can pick the one you want and develop faster.

File: settings.gradle [hunk 2]
```diff
@@ -0,0 +1,7 @@
+include ':fxa-client-library'
+include ':logins-library'
+include ':logins-sample'
+
+project(':fxa-client-library').projectDir = new File("fxa-client/sdks/android/library")
```
Comment (vladikoff) [inline settings.gradle]: Can we add docs somewhere on how to build these? 

For example I open Android Studio and it finds the 3 projects, but I know others use the CLI

File: scripts/taskcluster-android.sh [hunk 3]
```diff
@@ -2,56 +2,4 @@
...
-fi
-
-echo "Building selected targets: ${selected_targets[@]}."
-
-cd /build/application-services
-
-ORIG_PATH="$PATH"
-for target in "${selected_targets[@]}"
-do
-  PATH="$ANDROID_NDK_TOOLCHAIN_DIR/$target-$ANDROID_NDK_API_VERSION/bin:$ORIG_PATH"
```
Comment (vladikoff) [inline scripts/taskcluster-android.sh]: How does `$ANDROID_NDK_TOOLCHAIN_DIR` work now? Is it still used?

Comment (eoger) [reply scripts/taskcluster-android.sh]: It is set by https://github.com/mozilla/application-services/blob/9a2a57d2f1b681a60af8000b2e8758c4de6cfcc8/libs/android_defaults.sh and still used I believe.

Comment (ncalexan) [reply scripts/taskcluster-android.sh]: Correct -- it's magic that `rust-android-gradle` plugin knows about.  One more thing to document!

File: settings.gradle (hunk 2 above)
Comment (ncalexan) [reply settings.gradle]: Happy to, although this is pretty standard Gradle stuff.  I'll put it in a README somewhere.

**Comment by ncalexan** [comment]
> Wrong tcsc - not sure who you men, but probably not me!

Sorry, you are absolutely correct.  Damn mismatched handles!

File: fxa-client/sdks/android/library/build.gradle (hunk 1 above)
Comment (ncalexan) [reply fxa-client/sdks/android/library/build.gradle]: Re: uncommenting -- this is in a DO NOT LAND commit.

Re: separate tasks, I have been thinking about how to achieve this.  I have Vague Ideas about how to do this by hooking into the Android-Gradle plugin, and will try to dig in ASAP.

Comment (thomcc) [reply fxa-client/sdks/android/library/build.gradle]: I'd love to see how this is done with gradle if so. I couldn't figure out how to do things conditionally in any way in the cargo block except for stuff in `exec`...

Comment (ncalexan) [reply fxa-client/sdks/android/library/build.gradle]: Oh yeah, there's no uniform support for this at all.  If you look at the `rust-android-gradle` plugin, I did separate the `cargoBuild{Arm,Arm64,X86}` tasks; but they're not conditionally wired into the Android-Gradle plugin task tree.  That's non-trivial but clearly something that consumers (including me!) want, so I'll try to investigate it.

**Pull request merged**